### PR TITLE
fix: CL accumulator and tick initialization logic

### PR DESCRIPTION
**Pull request by czarcas7ic**
## What is the purpose of the change

This fixes two issues that were being observed:

1. We were reclaiming rewards that were already claimed when swapping in multiple directions
2. FeeGrowthOutside was not being recorded properly

## Brief Changelog

- Removes `validateAccumulatorValue` since the validation logic is no longer valid
  - Removes `accum_test.go` logic that relied on this validation 
- Initializes `feePositionAccumulator` to `the current feeAccumulator - the current feeGrowthOutside`
- When we update the `feeAccumulatorPosition`, we now first prep the initialFeeAccumulatorValue to `initialFeeAccumulatorValue + feeGrowthOutside`
  - Then, we update the unclaimed rewards and set the position's accumulator to the `feeAccumulator - feeGrowthOutside`
- Checks if the liquidityBefore is zero before updating the FeeGrowthOutside  

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/fees.go [hunk 1]
```diff
@@ -202,9 +224,16 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
...
-	// We need to update the position's accumulator to the current fee growth outside
-	// before we claim rewards.
-	if err := feeAccumulator.SetPositionCustomAcc(positionKey, feeGrowthOutside); err != nil {
+	position, err := accum.GetPosition(feeAccumulator, positionKey)
+	if err != nil {
+		return sdk.Coins{}, err
+	}
+
+	// prior to calling ClaimRewards (which updates unclaimed rewards), we must
+	// set initialFeeAccumulatorValue to initialFeeAccumulatorValue + feeGrowthOutside
```
Comment (p0mvn) [inline x/concentrated-liquidity/fees.go:233]:
```suggestion
	// set the position's accumulator value to the sum of
	// - the fee growth inside at position creation time (position.InitAccumValue)
	// - fee growth outside at the current block time (feeGrowthOutside)
```

File: x/concentrated-liquidity/fees.go [hunk 2]
```diff
@@ -202,9 +224,16 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
...
+	customAccumulatorValue := position.InitAccumValue.Add(feeGrowthOutside...)
+	err = feeAccumulator.SetPositionCustomAcc(positionKey, customAccumulatorValue)
+	if err != nil {
 		return sdk.Coins{}, err
 	}
```
Comment (p0mvn) [inline x/concentrated-liquidity/fees.go:238]: Let's encapsulate this into a function with tests please

File: x/concentrated-liquidity/tick.go [hunk 3]
```diff
@@ -30,6 +30,19 @@ func (k Keeper) initOrUpdateTick(ctx sdk.Context, poolId uint64, tickIndex int64
...
+	if liquidityBefore.IsZero() {
+		if tickIndex <= currentTick {
+			accum, err := k.getFeeAccumulator(ctx, poolId)
+			if err != nil {
+				return err
+			}
+
+			tickInfo.FeeGrowthOutside = accum.GetValue()
+		}
+	}
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick.go:44]: Let's cover this logic with a test please

File: x/twap/types/genesis.go [hunk 4]
```diff
@@ -82,8 +82,8 @@ func (t TwapRecord) validate() error {
 		return fmt.Errorf("twap record p1 accumulator cannot be negative, was (%s)", t.P1ArithmeticTwapAccumulator)
 	}
 
-	if t.GeometricTwapAccumulator.IsNil() || t.GeometricTwapAccumulator.IsNegative() {
-		return fmt.Errorf("twap record geometric accumulator cannot be negative, was (%s)", t.GeometricTwapAccumulator)
+	if t.GeometricTwapAccumulator.IsNil() {
+		return fmt.Errorf("twap record geometric accumulator cannot be nil, was (%s)", t.GeometricTwapAccumulator)
```
Comment (p0mvn) [inline x/twap/types/genesis.go:86]: Seems like merge conflict

File: x/twap/types/genesis_test.go [hunk 5]
```diff
@@ -73,6 +73,11 @@ func TestGenesisState_Validate(t *testing.T) {
 			})
 	)
 
+	withGeometricAcc := func(record TwapRecord, geometricAcc sdk.Dec) TwapRecord {
```
Comment (p0mvn) [inline x/twap/types/genesis_test.go:76]: This file has a conflict as well

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/fees.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/fees.go]: https://github.com/osmosis-labs/osmosis/pull/4265/commits/59f795f6e0af364976af773c4c70025c6d2a1475

File: x/concentrated-liquidity/tick.go (hunk 3 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/tick.go:44]: https://github.com/osmosis-labs/osmosis/pull/4265/commits/9e749648ad3bb9d1b9feb6c5ffb5207cef1dcb75

**Comment by czarcas7ic** [comment]
The only thing that needs to be updated now is the python script and we can then confirm it matches with the test errors here! CC: @p0mvn

**Pull request merged**